internal/resolution/variablesources/bundle_test.go

```diff
@@ -374,7 +563,7 @@ var _ = Describe("BundlesAndDepsVariableSource", func() {
 		)
 		_, err := bdvs.GetVariables(context.TODO())
 		Expect(err).To(HaveOccurred())
-		Expect(err.Error()).To(ContainSubstring("could not determine dependencies for bundle with id 'fake-catalog-test-package-bundle-2': could not find package dependencies for bundle 'bundle-2'"))
+		Expect(err.Error()).To(ContainSubstring(`could not determine dependencies for bundle with id "fake-catalog-test-package-bundle-2": could not find package dependencies for bundle`))
```

Comment:
While a savvy user that has memorized how to disambiguate the bundle ID into the separate parts (let's hope nobody puts a dash in the catalog name, package name or bundle name ...), all other users would prefer the more explicit error message you removed. It might be better to simply fix the top-level wrapper here to break up the bundle ID into constituent parts (is the transform lossy?).

Reply:
+1

Also, sort of related: I've noticed we don't really have a standard for how to build variable IDs. Another separate structural issue to follow-up on is to maybe write up a developer doc and/or write and ubiquitously use a helper function that can build/validate ID formats.

That could go a long way in making the bundle IDs more meaningful.

Reply:
I removed the part which was ambiguous: there could be multiple bundles with the same name from different catalogs/packages.

But I gree with you - this error message is not great. We need something more human readable. Good news is that we don't need to split bundle ID back into components: we have all the data in our structs.

I think what I'll do - I'll revert this change and deal with this in a separate PR.